### ISSUE TITLE
Cant "make" on CentOs main version

### ISSUE THREAD

**Issue by mcarrier69** [vysheng/tg#392]
Get this error ->
"make: *** No rule to make target `tgl/scheme.tl', needed by `auto/scheme.tl'.  Stop."

**Comment by mcarrier69** [comment]
git clone --recursive, will solve this issue above, but leads to -> this new issue -> 

[root@chi18-415-22 tg]# make
gcc -I. -I. -g -O2  -I/usr/local/include -I/usr/local/include -I/usr/local/include    -DHAVE_CONFIG_H -Wall -Wextra -Werror -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -c -MP -MD -MF dep/loop.d -MQ objs/loop.o -o objs/loop.o loop.c
loop.c: In function ‘accept_incoming’:
loop.c:624: error: ‘BEV_OPT_CLOSE_ON_FREE’ undeclared (first use in this function)
loop.c:624: error: (Each undeclared identifier is reported only once
loop.c:624: error: for each function it appears in.)
make: *** [objs/loop.o] Error 1

**Comment by vysheng** [comment]
try now

**Comment by mcarrier69** [comment]
works now, thank you!

**Comment by sagitter** [comment]
In addition to Fedora, telegram-cli is packaged on EPEL6/7 repositories for RHEL/CentOS: http://dl.fedoraproject.org/pub/epel/7/x86_64/repoview/telegram-cli.html

**Comment by danielberlin** [comment]
Antonio, in http://dl.fedoraproject.org/pub/epel/7/x86_64/repoview/telegram-cli.html *still* server.pub is installed in the wrong place:

It should be installed in ```/etc/telegram-cli/server.pub```, however it is wrongly installed in ```/etc/telegram-cli/server.pub/tg-server.pub```!

**Comment by sagitter** [comment]
What file .pub is the public key? server.pub or tg-server.pub?

Release note say: "By default public key is stored in the same folder named tg-server.pub or in /etc/telegram-cli/server.pub"; if public key is installed in /etc/telegram-cli/server.pub directory, i obtain:

```
$ telegram-cli
Telegram-cli version 1.2.0, Copyright (C) 2013-2015 Vitaly Valtman
Telegram-cli comes with ABSOLUTELY NO WARRANTY; for details type `show_license'.
This is free software, and you are welcome to redistribute it
under certain conditions; type `show_license' for details.
Telegram-cli uses libtgl version 1.2.0
I: config dir=[/home/sagitter/.telegram-cli]
[/home/sagitter/.telegram-cli] created
[/home/sagitter/.telegram-cli/downloads] created
 *** No public keys found
```

**Comment by danielberlin** [comment]
First of all, I think the sentence "By default public key is stored in the same folder named tg-server.pub or in /etc/telegram-cli/server.pub" has to be interpreted as:

Look for the public key either in ```./tg-server.pub``` or ````/etc/telegram-cli/server.pub``` (as a file, not a dir in which a tg-server.pub is kept). The sentence should better be: "By default, the public key is stored in tg-server.pub in the same folder or in /etc/telegram-cli/server.pub".

But why is it not working for you? That's funny! I compiled a fresh telegram-cli (GIT 1.2.0) on another CentOS 7 now and having ```/etc/telegram-cli/server.pub/tg-server.pub```, ```telegram-cli``` without ```-k``` gives:

```
change_user_group: can't find the user telegramd to switch to
Telegram-cli version 1.2.0, Copyright (C) 2013-2015 Vitaly Valtman
Telegram-cli comes with ABSOLUTELY NO WARRANTY; for details type `show_license'.
This is free software, and you are welcome to redistribute it
under certain conditions; type `show_license' for details.
Telegram-cli uses libtgl version 1.2.0
I: config dir=[/dev/.telegram-cli]
 *** No public keys found
```

However, storing server.pub correctly as ```/etc/telegram-cli/server.pub``` and running ```telegram-cli``` works.

**Comment by sagitter** [comment]
I'm packaging (RPM for Fedora/CentOS/RHEL) telegram-cli files in this way
```
/etc/telegram-cli
/etc/telegram-cli/server.pub
/etc/telegram-cli/tg-server.pub
/etc/telegram-cli/tg-server.pub/tg-server.pub
/usr/bin/telegram-cli
/usr/share/doc/telegram-cli
/usr/share/doc/telegram-cli/CHANGELOG
/usr/share/doc/telegram-cli/README-LUA
/usr/share/doc/telegram-cli/README.md
/usr/share/licenses/telegram-cli
/usr/share/licenses/telegram-cli/LICENSE
/usr/share/man/man8/telegram-cli.8.gz
```

telegram-cli works even if it seems not localize tg-server.pub:
```
$ telegram-cli -v
Telegram-cli version 1.2.0, Copyright (C) 2013-2015 Vitaly Valtman
Telegram-cli comes with ABSOLUTELY NO WARRANTY; for details type `show_license'.
This is free software, and you are welcome to redistribute it
under certain conditions; type `show_license' for details.
Telegram-cli uses libtgl version 1.2.0
I: config dir=[/home/sagitter/.telegram-cli]
[/home/sagitter/.telegram-cli] created
[/home/sagitter/.telegram-cli/downloads] created
 *** Couldn't open public key file: tg-server.pub
 *** Can not load key tg-server.pub
 *** adjusting CLOCK_MONOTONIC delta to -0.669181
 *** adjusting CLOCK_MONOTONIC delta to -0.566171
 *** adjusting CLOCK_MONOTONIC delta to 0.517465
 *** adjusting CLOCK_MONOTONIC delta to 0.654777
 *** adjusting CLOCK_MONOTONIC delta to -0.142110
phone number: 
```

I think public-key files should not be named like directories which contain them.

**Comment by danielberlin** [comment]
These are unnecessary:

```
/etc/telegram-cli/server.pub
/etc/telegram-cli/tg-server.pub
/etc/telegram-cli/tg-server.pub/tg-server.pub
```

Instead, there should be a single key file named server.pub inside the path /etc/telegram-cli:

```/etc/telegram-cli/server.pub```

See README.md
